src/odemis/driver/xt_client.py

```diff
@@ -1120,11 +1120,8 @@ def __init__(self, name, role, parent, hfw_nomag, **kwargs):
         pxs_scaled = (pxs[0] * self.scale.value[0], pxs[1] * self.scale.value[1])
         self._metadata[model.MD_PIXEL_SIZE] = pxs_scaled
 
-        self.scanMode = model.VAEnumerated(
-                self.parent.get_scan_mode(),
-                setter=self.parent.set_scan_mode,
-                choices={"external", "full_frame", "spot", "line"}
-        )
+        emode = self._getExternal()
+        self.external = model.BooleanVA(bool(emode), setter=self._setExternal)
```

Comment:
Let's drop the `bool()`, as that the only type that self._getExternal() returns. Maybe you could rename is to `_isExternal()` to avoid confusing with the VA getter. (But that's a little nitpicking)

Reply:
I don't understand your sentence. What are you suggesting to rename to `_isExternal()`?

Reply:
Indeed, I forgot an import word in my sentence :see_no_evil: . I meant rename `_getExternal()` to ` _isExternal()`. At least, it'd make it even clearer that it returns a bool.